### PR TITLE
Updated documentation how to debug Jest tests in IntelliJ

### PR DESCRIPTION
**Pull request by acierto**
Signed-off-by: bogdannechyporenko <bogdannechiporenko@gmail.com>

## Hey, I just made a Pull Request!

I'd love to debug the jest tests from IDE.
There are few missing tweaks required for that.
Updated also a documentation how to configure it locally. 
I can also attach a screenshot to the documentation, but didn't do it yet, as no screenshots were used in the doc (so I only added a description of it).

Prove that it really works:

Screenshot 2023-01-29 at 14 12 28

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

We won't switch to `ts-jest`, but good news is that we don't need to :grin:

I've switched to inline source maps over in #16033. It works fine for debugging in VSCode, could you try that change out in IntelliJ and we can update docs accordingly? 

This is all documented over at https://backstage.io/docs/local-dev/cli-build-system#jest-configuration btw, I suggest we update the docs there if needed and point the plugin testing docs there instead.

**Comment by acierto** [comment]
Hey @Rugvip !

Checked your change, it works too.
Didn't realise that `sourceMaps` there could also affect debugging capabilities. 

Do you want me to update my PR to move the docs to a new location or will you do it in your PR?

**Comment by acierto** [comment]
As your PR was already merged by @benjdlambert, left in this PR only docs for IntelliJ users and moved it to another location, as you suggested.

File: .changeset/selfish-rabbits-compare.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': minor
+---
+
+Fixing Jest configuration to be able to debug tests via IDE
```
Comment (freben) [inline .changeset/selfish-rabbits-compare.md:5]: This file should be removed now, and you should revert the entire yarn.lock as well

File: docs/local-dev/cli-build-system.md [hunk 2]
```diff
@@ -595,6 +595,33 @@ If you also want to enable source maps when debugging tests, you can do so by se
 }
 ```
 
+### Debugging Jest Tests
+
+To configure the debugging of your tests in IntelliJ, you have to:
+
+1. Enable in the root `tsconfig.json`
```
Comment (freben) [inline docs/local-dev/cli-build-system.md:602]: this whole point is not needed now that we have inline source maps

File: docs/local-dev/cli-build-system.md [hunk 3]
```diff
@@ -595,6 +595,33 @@ If you also want to enable source maps when debugging tests, you can do so by se
...
+
+- Click on "Edit Configurations" on top panel
+- Click "+"
+- And choose there "Jest"
+
+3. Fill in the next fields as:
+
+- Jest package: ~/proj/backstage/node_modules/jest
+- Working directory: (it depends on where is your test located, i.e. ~/proj/backstage/packages/core-plugin-api)
+- Jest config: --config ../../packages/cli/config/jest.js --runInBand
```
Comment (freben) [inline docs/local-dev/cli-build-system.md:623]: Doesn't intellij have a way of dynamically saying "the root of the project"? Like a `%` or `$` variable or something? It's not great to have literally `~ /proj` in the docs, or that `../../` relative path which feels fragile.

Did you really have to point to the exact package in the second point? That's gonna be really annoying to have to change what package it points to every single time you want to run tests in a different package.

Comment (acierto) [reply docs/local-dev/cli-build-system.md:623]: Hi @freben thank you for your comments. 
Really valid points, let me describe another approach, hope you'll like it

Comment (acierto) [reply docs/local-dev/cli-build-system.md]: Updated the doc.

It's possible to update the Jest template once

Screenshot 2023-01-31 at 11 49 10

IntelliJ doesn't like variables in that modal 🤷 

Screenshot 2023-01-31 at 11 52 13

**Pull request closed without merging**